tree/ntuple/v7/src/RField.cxx

```diff
@@ -916,6 +923,181 @@ void ROOT::Experimental::RClassField::AcceptVisitor(Detail::RFieldVisitor &visit
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RCollectionClassField::RCollectionClassField(std::string_view fieldName, std::string_view className)
+   : RCollectionClassField(fieldName, className, TClass::GetClass(std::string(className).c_str()))
+{
+}
+
+ROOT::Experimental::RCollectionClassField::RCollectionClassField(std::string_view fieldName, std::string_view className,
+                                                                 TClass *classp)
+   : ROOT::Experimental::Detail::RFieldBase(fieldName, className, ENTupleStructure::kCollection, false /* isSimple */),
+     fNWritten(0)
+{
+   if (classp == nullptr)
+      throw RException(R__FAIL("RField: no I/O support for collection proxy type " + std::string(className)));
+
+   fProxy = classp->GetCollectionProxy();
```

Comment:
Currently the CollectionProxy are stateful (it is planned to remove this) and not thread safe.  To alleviate this each user needs its own copy of the collection proxy.
```suggestion
   fProxy = classp->GetCollectionProxy()->Generate();
```